example_docs/api_example/electron.py

```diff
@@ -33,3 +36,38 @@ def compute_momentum(self, velocity: float) -> float:
     
     def method_with_a_reallyyreallreallyreallyreallyreallyreallreallyreallyreallyreally_long_title(self):
         """blahblahblahblahblahblahblahblahblahblahblahblahblahblahblahblahblahblah"""
+
+    @overload
+    def overloaded_func(
+        self,
+        arg1: tuple[str, str],
+        arg2: list[str],
+        arg3: int,
+        arg4: Electron,
+    ) -> None:
+        ...
+
+    @overload
+    def overloaded_func(
+        self,
+        arg1: tuple[int, int],
+        arg2: list[int],
+        arg3: bool,
+        arg4: set[Electron],
+    ) -> None:
+        ...
+
+    def overloaded_func(
+        self,
+        arg1: tuple[str, str] | tuple[int, int],
+        arg2: list[str] | list[int],
+        arg3: int | bool,
+        arg4: Electron | set[Electron],
+    ) -> None:
+        """This is meant to test out https://github.com/Qiskit/qiskit_sphinx_theme/pull/319.
+
+        :param arg1: Tuples ftw!
```

Comment:
Could you use the same params syntax that we use in the qiskit docstrings please?

e.g. https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/circuit/library/graph_state.py#L59-L70

Reply:
What's the motivation? The final docs build doesn't matter whether we use Napoleon vs. the Sphinx standard. Napoleon converts the Qiskit-style docstrings into this standard style that I'm using here, before it ever gets to our qiskit_sphinx_theme code/styling.

Reply:
Yeah I guess its not a big deal, I just thought we were trying to be consistent with what terra does, but I'm not clear where that line is. It feels very arbitrary atm which things we decide to align and which things we don

Reply:
> I just thought we were trying to be consistent with what terra does, but I'm not clear where that line is.

Yeah, it's true that we in general do want to align unless we have a good reason not to (e.g. Pytest).

Here, my only reason not to align is "I'm lazy and don't want to rewrite this file" hahaha. So, I'll fix it now (or ask ChatGPT to lol).